python/pyiceberg/transforms.py

```diff
@@ -269,13 +270,9 @@ def hash_func(v: Any) -> int:
         elif source_type == UUIDType:
 
             def hash_func(v: Any) -> int:
-                return mmh3.hash(
-                    struct.pack(
-                        ">QQ",
-                        (v.int >> 64) & 0xFFFFFFFFFFFFFFFF,
-                        v.int & 0xFFFFFFFFFFFFFFFF,
-                    )
-                )
+                if isinstance(v, UUID):
+                    return mmh3.hash(v.bytes)
+                return mmh3.hash(v)
```

Comment:
I found another place that could be simplified based on the above comment. Also, I think we need to let the hash function handles `bytes` since `transform` usually calls on literal's value and now UUIDLiteral stores `bytes`

Reply:
Awesome, thanks for fixing this, much better 👍🏻